File: backend/ai/views/ai_api_key.py
from ai.models import AIApiKey
from utils.serializers import CustomModelSerializer
from utils.custom_model_viewSet import CustomModelViewSet

class AIApiKeySerializer(CustomModelSerializer):
    """
    AI API 密钥 序列化器
    """
    class Meta:
        model = AIApiKey
        fields = '__all__'
        read_only_fields = ['id', 'create_time', 'update_time']


class AIApiKeyViewSet(CustomModelViewSet):
    """
    AI API 密钥 视图集
    """
    queryset = AIApiKey.objects.filter(is_deleted=False).order_by('-id')
    serializer_class = AIApiKeySerializer
    filterset_fields = ['id', 'remark', 'creator', 'modifier', 'is_deleted', 'name', 'platform', 'api_key', 'url', 'status']
    search_fields = ['name']  # 根据实际字段调整
    ordering_fields = ['create_time', 'id']
    ordering = ['-create_time']

# 移入urls中
#

File: web/apps/web-antd/src/models/ai/ai_api_key.ts
import { BaseModel } from '#/models/base';

export namespace AiAIApiKeyApi {
  export interface AiAIApiKey {
    id: number;
    remark: string;
    creator: string;
    modifier: string;
    update_time: string;
    create_time: string;
    is_deleted: boolean;
    name: string;
    platform: string;
    model_type: string;
    api_key: string;
    url: string;
    status: number;
  }
}

export class AiAIApiKeyModel extends BaseModel<AiAIApiKeyApi.AiAIApiKey> {
  constructor() {
    super('/ai/api_key/');
  }
}
